core/capabilities/ccip/ccipaptos/commitcodec.go

```diff
@@ -0,0 +1,192 @@
+package ccipaptos
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/aptos-labs/aptos-go-sdk"
+	"github.com/aptos-labs/aptos-go-sdk/bcs"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// CommitPluginCodecV1 is a codec for encoding and decoding commit plugin reports.
+// Compatible with ccip::offramp version 1.6.0
+type CommitPluginCodecV1 struct{}
+
+var _ cciptypes.CommitPluginCodec = (*CommitPluginCodecV1)(nil)
+
+func NewCommitPluginCodecV1() *CommitPluginCodecV1 {
+	return &CommitPluginCodecV1{}
+}
+
+func (c *CommitPluginCodecV1) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
+	s := &bcs.Serializer{}
+	bcs.SerializeSequenceWithFunction(report.PriceUpdates.TokenPriceUpdates, s, func(s *bcs.Serializer, item cciptypes.TokenPrice) {
+		sourceToken := aptos.AccountAddress{}
+		err := sourceToken.ParseStringRelaxed(string(item.TokenID))
+		if err != nil {
+			s.SetError(fmt.Errorf("failed to parse source token address: %w", err))
+			return
+		}
+		s.Struct(&sourceToken)
+		if item.Price.IsEmpty() {
+			s.U256(*big.NewInt(0))
```

Comment:
Have we seen this code path get executed in practice? Seems like it should be an error but not 100% sure.